lib/CMakeLists.txt

```diff
@@ -0,0 +1,159 @@
+
+set(use_math_DEFS "")
+if(MSVC)
+  set(use_math_DEFS "-D_USE_MATH_DEFINES=1")
+endif()
+
+build_library_in_subdir(datetime)
+
+add_subdirectory(gis)
+
+build_library_in_subdir(driver
+  DEFS "${use_math_DEFS}"
+  DEPENDS grass_gis FREETYPE ICONV ZLIB)
+
+add_subdirectory(proj)
+
+file(GLOB raster_SRCS "./raster/*.c")
+list(APPEND raster_SRCS "./gis/gisinit.c")
```

Comment:
`gisinit.c` shouldn't be included in other translation units (TU) other than than the gislib. The G__ struct is supposed to be external linkage wrt to other libraries. There will be issues on MSVC like some unrecoginized symbol. But the right fix is to have declspec(import/export) on the G__  global struct.